Add Fix Invalid Cell and Clear All actions to `GridMap`

simpuid opened the pull request:
Added option to fix invalid cells in `GridMap` just like `TileMap`.
Also added an option to Clear all cells of `GridMap`, this fixes/resolve #37124 
Both actions are working properly, even with undo-redo.

akien-mga commented:
What if you want to swap the MeshLibrary to another, compatible one (i.e. different meshes but with the same logical use at the same IDs)? I do this a lot with TileMaps (see e.g. https://github.com/KOBUGE-Games/jetpaca where different TileSets use the exact same layout).

simpuid commented:
Then that would be a problem. Maybe we can clear the internal data only when mesh library is empty?

Alastor01 commented:
> What if you want to swap the MeshLibrary to another, compatible one (i.e. different meshes but with the same logical use at the same IDs)? I do this a lot with TileMaps (see e.g. https://github.com/KOBUGE-Games/jetpaca where different TileSets use the exact same layout).

Would it be possible to have two options? As there can be two scenarios:

A) Updating to a compatible library, like you said

B) Changing library to a different one with incompatibles IDs - in this case not removing old references would make a mess of a map (an example would be having library with new tiles or missing ones)

Commit:
Added functionalities to fix invalid cells and clear all cells in GridMap

Changes made:
* Added condition to `GridMap::set_mesh_library` to update only if new library is different from old one
* Added `GridMap::fix_invalid_cells` taking inspiration from `TileMap::fix_invalide_tiles`
* Added menu entry to clear cells in `GridMapEditor`
* Added menu entry to fix tiles in `GridMapEditor`

simpuid commented:
@Alastor01 that would cause problem when user will undo the mesh library change. The cell data will remain empty even after undoing it.
One of the solution is to adapt it like `TileMap` :
`TileMap` has function to fix invalid tiles.
![gridmap](https://user-images.githubusercontent.com/37181756/77032049-65791b00-69c9-11ea-822c-0ae3fbc4c2da.png)
I have implemented the same functionality and also added an option to clear all cells.
![gridmap_new](https://user-images.githubusercontent.com/37181756/77032325-17184c00-69ca-11ea-9e04-87803a2cb406.png)
Things are working properly even undo-redo.
@akien-mga what are your opinions? Should I open a new PR for that or force push to this one?

Alastor01 commented:
@simpuid That's interesting, haven't used 2D part of the engine for ages.

This looks great, thank you!

akien-mga commented:
Sorry for the delay reviewing this. The feature seems to make sense, it should maybe be synced with how this is handled in the TileMap editor in 4.0 to be consistent (which is also able to display invalid cells in the viewport).
